app/services/check_course_jobs.rb

```diff
@@ -30,21 +48,43 @@ def delete_unique_lock
   def expected_digest
     hash = {
       'class' => 'CourseDataUpdateWorker',
-      'queue' => queue_for(@course),
+      'queue' => @queue,
       'unique_args' => [@course_id]
     }.to_json
     digest = OpenSSL::Digest::MD5.hexdigest hash
     "uniquejobs:#{digest}"
   end
 
-  def find_job
+  def find_job # rubocop:disable Metrics/CyclomaticComplexity
+    Sidekiq::ScheduledSet.new.select do |retri|
+      next unless retri.klass == 'CourseDataUpdateWorker'
+      return retri if retri.args == [@course_id]
+    end
+
     Sidekiq::Queue.all.each do |queue|
       queue.each do |job|
         next unless job.klass == 'CourseDataUpdateWorker'
         return job if job.args == [@course_id]
       end
     end
 
+    Sidekiq::Workers.new.each do |_process_id, _thread_id, work|
+      next unless work['payload']['class'] == 'CourseDataUpdateWorker'
+      return work if work['payload']['args'] == [@course_id]
+    end
+
     return nil
   end
+
+  def delete_orphan_lock
+    if find_job.nil? && SidekiqUniqueJobs::Digests.all.include?(expected_digest)
```

Comment:
The check for whether it's locked can be extracted to its own method, and you can add a method like `job_exists?` that does `find_job.nil?`.

Reply:
Done!